sdk/template/aztemplate/client.go

```diff
@@ -1,37 +1,53 @@
-//go:build go1.16
-// +build go1.16
+//go:build go1.18
+// +build go1.18
 
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
 package aztemplate
 
 import (
+	"context"
+
 	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/runtime"
 	"github.com/Azure/azure-sdk-for-go/sdk/template/aztemplate/internal"
 )
 
 // ClientOptions contains optional parameters for NewClient
-type ClientOptions struct{}
+type ClientOptions struct {
+	azcore.ClientOptions
+}
 
-// Client is the client to interact with
+// Client is the client to interact with.
+// Don't use this type directly, use NewClient() instead.
 type Client struct {
 	client *internal.TemplateClient
 }
 
 // NewClient returns a pointer to a Client
 func NewClient(cred azcore.TokenCredential, options *ClientOptions) (*Client, error) {
+	if options == nil {
+		options = &ClientOptions{}
+	}
 
-	//if options == nil {
-	//options = &ClientOptions{}
-	//}
-	tc := internal.NewTemplateClient()
+	pl := runtime.NewPipeline(moduleName, moduleVersion, runtime.PipelineOptions{
+		PerRetry: []policy.Policy{
+			runtime.NewBearerTokenPolicy(cred, []string{"service_scope"}, nil),
+		},
+	}, &options.ClientOptions)
 
-	return &Client{client: tc}, nil
+	return &Client{client: internal.NewTemplateClient(pl)}, nil
+
+}
 
+// SomeServiceActionOptions contains the optional values for the Client.SomeServiceAction method.
+type SomeServiceActionOptions struct {
+	// placeholder for future options
```

Comment:
Should this demonstrate when fields should be pointers to things (stuff that goes on the wire) and when they should be the things themselves (stuff that does not like resume tokens)?

Reply:
Good point.  In fact, the exiting API needs further refinement to return a response type as well as an error.  And if we were to really flesh this out, examples for pagers and pollers should be added.